Support Date

## Describe the bug
camelizeDeep doesn't support a Date object.

## Steps to Reproduce
The following test is failed. camelizeDeep converts a Date instance to empty object.

```typescript
// date_test.ts
import { assertEquals } from "https://deno.land/std@0.127.0/testing/asserts.ts";
import { camelizeDeep } from "./mod.ts";

Deno.test("camelizeDeep: Object with Date instance", () => {
  assertEquals(
    camelizeDeep({
      id: "test_123",
      created_at: new Date("2022-01-31T12:00:00.000"),
    }),
    {
      id: "test_123",
      createdAt: new Date("2022-01-31T12:00:00.000"),
    },
  );
});
```

```sh
~/P/ts-camelize ❯❯❯ deno test date_test.ts
Check file:///Users/ioriikeda/Projects/ts-camelize/date_test.ts
running 1 test from file:///Users/ioriikeda/Projects/ts-camelize/date_test.ts
test camelizeDeep: Object with Date instance ... FAILED (15ms)

failures:

camelizeDeep: Object with Date instance
AssertionError: Values are not equal:


    [Diff] Actual / Expected


    {
-     createdAt: {},
-     id: "test123",
+     createdAt: 2022-01-31T03:00:00.000Z,
+     id: "test_123",
    }

    at assertEquals (https://deno.land/std@0.127.0/testing/asserts.ts:269:9)
    at file:///Users/ioriikeda/Projects/ts-camelize/date_test.ts:7:3
    at testStepSanitizer (deno:runtime/js/40_testing.js:432:13)
    at asyncOpSanitizer (deno:runtime/js/40_testing.js:145:15)
    at resourceSanitizer (deno:runtime/js/40_testing.js:360:13)
    at Object.exitSanitizer [as fn] (deno:runtime/js/40_testing.js:415:15)
    at runTest (deno:runtime/js/40_testing.js:673:18)
    at Object.runTests (deno:runtime/js/40_testing.js:786:28)
    at [deno:cli/tools/test.rs:512:6]:1:21

failures:

        camelizeDeep: Object with Date instance

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out (42ms)

error: Test failed
```

## Expected behavior
camelizeDeep should support Date

--- a/mod.ts
+++ b/mod.ts
@@ -6,29 +6,37 @@ export type Camelize<S extends string> = S extends
 
 export type CamelizeDeep<T> = T extends string ? Camelize<T>
   : T extends Array<infer U> ? Array<CamelizeDeep<U>>
-  : // deno-lint-ignore no-explicit-any
-  T extends Record<string, any>
+  : T extends Record<string, unknown>
     ? { [K in keyof T as Camelize<K & string>]: CamelizeDeep<T[K]> }
   : T;
+
+// Determines an object is a class object or not.
+const isPlainObject = (
+  obj: unknown,
+): obj is Record<string | symbol | number, unknown> => {
+  const prototype = Object.getPrototypeOf(obj);
+  return typeof obj === "object" &&
+    (prototype === null || prototype.constructor === Object);
+};
 
 export const camelize = (str: string) => {
   return str.replace(/[_.-](\w|$)/g, (_, x) => x.toUpperCase());
 };
 
-// deno-lint-ignore no-explicit-any
-export const camelizeDeep = <T extends string | Record<string, any>>(
+export const camelizeDeep = <T extends unknown>(
   obj: T,
 ): T extends string ? string : CamelizeDeep<T> => {
   return typeof obj === "string"
     ? camelize(obj)
-    : typeof obj !== "object"
-    ? obj
     : Array.isArray(obj)
     ? obj.map(camelizeDeep)
-    : Object.keys(obj).reduce((acc, key) => {
-      const camelizedKey = camelize(key);
-      acc[camelizedKey] = camelizeDeep(obj[key]);
-      return acc;
-      // deno-lint-ignore no-explicit-any
-    }, {} as any);
+    : // NOTE: Eliminate class instances like Date, Set, RegExp, etc.
+      !isPlainObject(obj)
+      ? obj
+      : Object.keys(obj).reduce((acc, key) => {
+        const camelizedKey = camelize(key);
+        acc[camelizedKey] = camelizeDeep(obj[key]);
+        return acc;
+        // deno-lint-ignore no-explicit-any
+      }, {} as any);
 };

--- a/mod_test.ts
+++ b/mod_test.ts
@@ -30,32 +30,49 @@ type cases = [
 ];
 
 // Conversion Checks
-assertEquals(camelize("abc_def-ghi.j123"), "abcDefGhiJ123");
-assertEquals(
-  camelizeDeep({
-    user_id: 123,
-    bookings: [
-      {
-        booking_id: 1,
-        is_confirmed: true,
-      },
-      {
-        booking_id: 2,
-        is_confirmed: false,
-      },
-    ],
-  }),
-  {
-    userId: 123,
-    bookings: [
-      {
-        bookingId: 1,
-        isConfirmed: true,
-      },
-      {
-        bookingId: 2,
-        isConfirmed: false,
-      },
-    ],
-  },
-);
+Deno.test("camelize", () => {
+  const camelized = camelize("abc_def-ghi.j123");
+  assertEquals(camelized, "abcDefGhiJ123");
+});
+
+Deno.test("camelizeDeep: Object with Array", () => {
+  assertEquals(
+    camelizeDeep({
+      user_id: 123,
+      bookings: [
+        {
+          booking_id: 1,
+          is_confirmed: true,
+        },
+        {
+          booking_id: 2,
+          is_confirmed: false,
+        },
+      ],
+    }),
+    {
+      userId: 123,
+      bookings: [
+        {
+          bookingId: 1,
+          isConfirmed: true,
+        },
+        {
+          bookingId: 2,
+          isConfirmed: false,
+        },
+      ],
+    },
+  );
+});
+
+Deno.test("camelizeDeep: Object with Date instance", () => {
+  assertEquals(
+    camelizeDeep({
+      created_at: new Date("2022-01-31T12:00:00.000"),
+    }),
+    {
+      createdAt: new Date("2022-01-31T12:00:00.000"),
+    },
+  );
+});

